UN-OCHA/hpc-cdm
# HPC-9034 Update Nx 16, React 18 and respective libraries

## Describe your changes:

- 🚀 Upgraded Nx to v16.3.1
- 🚀 Upgraded React to v18.2.0
- 🚀 Upgraded Jest to v29.5.0 (Nx upgrade related)
- 🚀 Upgraded React Router to v6.11.2
- 📦  Different package updates
- Updated `<Switch>` React Router v5 component to `<Routes>` React Router v6
- Updated `<Route>` structure, React Router v6 has been thought to try to use relative paths (Didn't want to refactor too much to not go out of the task, but we could define how to write it correctly and that is easy to understand)
- Added `usePrompt()` hook to show prompts in the application when we need
 
Please refer to the commit messages to check a more in-detailed explanation of the changes step by step

## How can I test the changes?:

You should init the app and test [here](http://localhost:4200)
To test the changes is important to:

1. Check the navigation through the page and check that no link has been broken
2. Check that in the Enketo form of any operation, you can't leave an unsaved form without a prompt appearing (For reload action and closing tab action, the browser prompt should appear instead of ours)
3. The application loads correctly
4. Check package.json to see if something seems off

## TO DO:
I would say there is nothing left to do of the task

## Does any part of this code spark :star: joy :star:?:

No special or remarkable things were done I would say. (Except becoming a full-time reader of documentation)
